pg_backup_api/pg_backup_api/openapi_server/models/status.py

```diff
@@ -1162,20 +1162,30 @@ def postgresql_systemid(self, postgresql_systemid):
     def replication_slot(self):
         """Gets the replication_slot of this Status.
 
+        Details of the replication slot being used by Barman on this PostgreSQL server in the form of an ordered array. The first item MUST be a string containing the slot_name. The second item MUST be a boolean reflecting the active state of the replication slot. The third item MUST be a string containing the LSN of the oldest WAL which still might be required by the consumer of this slot.   # noqa: E501
 
         :return: The replication_slot of this Status.
-        :rtype: List
+        :rtype: List[object]
         """
         return self._replication_slot
 
     @replication_slot.setter
     def replication_slot(self, replication_slot):
         """Sets the replication_slot of this Status.
 
+        Details of the replication slot being used by Barman on this PostgreSQL server in the form of an ordered array. The first item MUST be a string containing the slot_name. The second item MUST be a boolean reflecting the active state of the replication slot. The third item MUST be a string containing the LSN of the oldest WAL which still might be required by the consumer of this slot.   # noqa: E501
 
         :param replication_slot: The replication_slot of this Status.
-        :type replication_slot: List
-        """
+        :type replication_slot: List[object]
+        """
+        if replication_slot is not None and len(replication_slot) > 3:
```

Comment:
Why use  two conditions and not test  `len(replication_slot) == 3` and display both expected and actual number of elements in error message?

Reply:
Just wondering: are those 3 elements prone to error? ie do we need to test them (string, bool, string) and maybe more complex test?

Reply:
Good questions.

This is openapi-generated code and it's verifying that the data matches the schema. There's no reason why the we *would* need more testing - in practice either all elements will be there or it will be null.

We could improve it but I don't want to have to dig too deeply into openapi/connexion at this point.

Reply:
Well if is is an automation no need to fight agains it (someone has probably already tested this).